__tests__/package-request.js

```diff
@@ -6,16 +6,25 @@ import PackageResolver from '../src/package-resolver.js';
 import Lockfile from '../src/lockfile';
 import Config from '../src/config.js';
 
-async function prepareRequest(pattern, version, resolved): Object {
+async function prepareRequest(pattern: string, version: string, resolved: string, parentRequest?: Object): Object {
   const privateDepCache = {[pattern]: {version, resolved}};
-  const lockfile = new Lockfile({cache: privateDepCache});
   const reporter = new reporters.NoopReporter({});
   const depRequestPattern = {
     pattern,
     registry: 'npm',
     hint: null,
     optional: false,
+    parentNames: [],
+    parentRequest,
   };
+  if (parentRequest) {
+    depRequestPattern.parentRequest = parentRequest;
+    depRequestPattern.parentNames = parentRequest.parentNames.slice(0);
```

Comment:
You can use `Array.from(parentRequest.parentNames)` instead ;)

Reply:
True!!!  Good Catch!! I was working on JS libraries that needed to support IE8 😢 😭 , so picked up a few habits there... Also, i think `Array.prototype.slice` is faster than `Array.from` https://jsperf.com/arrayclonetest1/1